src/_P039_Thermocouple.ino

```diff
@@ -259,34 +501,52 @@ float readMax6675(struct EventStruct *event)
 
 float readMax31855(struct EventStruct *event)
 {
-  // take the SS pin low to select the chip:
-  digitalWrite(Plugin_039_Get_SPI_CS_Pin(event), LOW);
+  union {
+      uint8_t messageBuffer[4] = {0x00, 0x00, 0x00, 0x00};
+      uint32_t value;
+  } u_mB;
 
-  // "transfer" 0x0 and read the MSB Data from the Chip
-  uint32_t rawvalue = SPI.transfer16(0x0);
+  uint8_t CS_pin_no = get_SPI_CS_Pin(event);
 
-  // Shift MSB 16 Bits to the left
-  rawvalue <<= 16;
+  // u_mB.messageBuffer[4] = {0x00, 0x00, 0x00, 0x00};
 
-  // "transfer" 0x0 and read the LSB Data from the Chip
-  rawvalue |= SPI.transfer16(0x0);
+  // "transfer" 0x0 and read the 32 Bit conversion register from the Chip
 
-  // take the SS pin high to de-select the chip:
-  digitalWrite(Plugin_039_Get_SPI_CS_Pin(event), HIGH);
+  transfer_n_ByteSPI(CS_pin_no, 4, &u_mB.messageBuffer[0]);
+  uint16_t rawvalue = u_mB.value;
 
-# ifndef BUILD_NO_DEBUG
+  // uint32_t rawvalue = read_n_ByteitRegister(CS_pin_no, 0x0000);
 
-  if (loglevelActiveFor(LOG_LEVEL_DEBUG)) {
-    String log = F("P039 : MAX31855 : RAW - BIN:");
+  // // Shift MSB 16 Bits to the left
+  // rawvalue <<= 16;
 
-    log += String(rawvalue, BIN);
-    log += " HEX:";
-    log += String(rawvalue, HEX);
-    log += " DEC:";
-    log += String(rawvalue);
-    addLog(LOG_LEVEL_DEBUG, log);
-  }
-# endif // ifndef BUILD_NO_DEBUG
+  // // "transfer" 0x0 and read the LSB Data from the Chip
+  // rawvalue |= read16BitRegister(CS_pin_no, 0x0000);
+
+  # ifndef BUILD_NO_DEBUG
+
+    if (loglevelActiveFor(LOG_LEVEL_DEBUG)) 
+    {
+      String log = F("P039 : MAX31855 : RAW - BIN:");
```

Comment:
When building long strings it's best to use `string.reserve(<size>)` to avoid several re-allocations. Size should be enough to hold the resulting string ofc.
```
String log;
log.reserve(64);
log = F("P039 : MAX31855 : RAW - BIN:");
...
```

Reply:
And to make it even better, you can test the result of a `reserve`, so you won't try to continue if there isn't room for the intended string anyway.

Reply:
That comment leaves me with a question mark - test dring runtime  when trying to reserve or during validation ?

Reply:
Finally got the comment - of course testing during runtime ! - Introduced in the code, with one of the next pushes